Microsoft.Toolkit/Diagnostics/Generated/Guard.Collection.cs

```diff
@@ -0,0 +1,1502 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/* ========================
```

Comment:
I checked them in for now because apparently VS only rebuilds them if you open one and save it, and not when you just build the solution. If there's an option for that then yeah it'd be possible not to check them in. Since I didn't find that though I've left them in so that people cloning the repo don't end up with either build errors or missing APIs because of that.
If it's possible to set that, I'm all ears! 😄

As for the outputs being in separate folders, I don't think that's necessary, since VS has a very nice display view for them, just like with `.xaml` and `.xaml.cs` files, so it bundles them together:

![image](https://user-images.githubusercontent.com/10199417/76370780-40850800-6338-11ea-9e54-2b57da0de2e4.png)

Reply:
Yeah, I'm not sure if there's a build action or not., good to know about the VS display, it's been a bit since I've worked with T4 templates too :)